theano/tensor/tests/test_opt.py

```diff
@@ -1895,18 +1889,17 @@ def test_stacktrace(self):
         # which requires us to add the 'canonicalize' phase.
         mode = theano.compile.mode.Mode(optimizer=None).including('canonicalize_db').including("local_subtensor_make_vector")
         f = function([x, y, z], v[0], mode=mode)
-        # Check stacktrace was copied over correctly after opt was applied
-        # FIXME: DeepCopyOp does not have a trace
-        #self.assertTrue(check_stack_trace(f, ops_to_check='all'))
-        
+        # The resulting graph only has a DeepCopyOp, for which the
+        # copy_stack_trace is not called. (See issue #4421)
+        # self.assertTrue(check_stack_trace(f, ops_to_check='all'))
         
         # Compile function using all optimizations in fast_compile mode, 
         # including the 'local_subtensor_make_vector' optimization
         mode = theano.compile.mode.get_mode('FAST_COMPILE').including("local_subtensor_make_vector")
         f = function([x, y, z], v[0], mode=mode)
-        # Check stacktrace was copied over correctly after opt was applied
-        # FIXME: DeepCopyOp does not have a trace
-        #self.assertTrue(check_stack_trace(f, ops_to_check='all'))
+        # The resulting graph only has a DeepCopyOp, for which the
+        # copy_stack_trace is not called. (See issue #4421)
+        # self.assertTrue(check_stack_trace(f, ops_to_check='all'))
         
```

Comment:
The issue here is not that the DeepCopyOp has no stack trace, it is that the two cases in this test do not check the case where `local_subtensor_make_vector` inserts a `Subtensor` node.
It is OK to just refer to #4421 for now and not fix it.